spec/tagManagerWrap.spec.js

```diff
@@ -0,0 +1,185 @@
+import TagManager from '../src/tagmanager-wrap';
+
+describe('Tagmanager', () => {
+  const tagManager = new TagManager([], {
+    gtmId: 'GTM-T2HGD',
+    startPush: {
+      page_type: 'pages:demo',
+      path_category: 'reformas-e-reparos/demo',
+      experiments: window.experiments || [],
+    },
+  });
+
+  describe('constructor', () => {
+    it('defines default properties', () => {
+      expect(tagManager.options).toBeDefined();
+      expect(tagManager.dataLayer).toBeDefined();
+    });
+  });
+
+  describe('.defineProperties', () => {
+    it('sets default properties', () => {
+      expect(tagManager.options.gtmId).toEqual('GTM-T2HGD');
+      expect(tagManager.options.startPush.page_type).toEqual('pages:demo');
+      expect(tagManager.options.startPush.path_category).toEqual('reformas-e-reparos/demo');
+      expect(tagManager.options.virtualPageViewEvent).toEqual('virtual_pageview');
+    });
+  });
+
+  describe('.dataLayer init', () => {
+    it('sets initial data', () => {
+      expect(tagManager.dataLayer.length).not.toEqual(0);
+    });
+  });
+
+  describe('.init', () => {
+    beforeEach(() => {
+      spyOn(tagManager, 'appendAsyncScript');
+      spyOn(tagManager, 'appendNoScriptFallBack');
+      tagManager.init();
+    });
+
+    it('calls appendAsyncScript', () => {
+      expect(tagManager.appendAsyncScript).toHaveBeenCalled();
+    });
+
+    it('calls appendNoScriptFallBack', () => {
+      expect(tagManager.appendNoScriptFallBack).toHaveBeenCalled();
+    });
+  });
+
+
+  describe('.prependExperiment', () => {
+    beforeEach(() => {
+      tagManager.init();
+      tagManager.prependExperiment({
```

Comment:
Here values have to be variable too?

Reply:
@williamcosta for the test we don't need it to be that dynamic 👍